single-pool/program/src/processor.rs

```diff
@@ -707,6 +707,64 @@ impl Processor {
         Ok(())
     }
 
+    fn process_reactivate_pool(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let vote_account_info = next_account_info(account_info_iter)?;
+        let pool_info = next_account_info(account_info_iter)?;
+        let pool_stake_info = next_account_info(account_info_iter)?;
+        let pool_stake_authority_info = next_account_info(account_info_iter)?;
+        let clock_info = next_account_info(account_info_iter)?;
+        let clock = &Clock::from_account_info(clock_info)?;
+        let stake_history_info = next_account_info(account_info_iter)?;
+        let stake_config_info = next_account_info(account_info_iter)?;
+        let stake_program_info = next_account_info(account_info_iter)?;
+
+        check_vote_account(vote_account_info)?;
+        check_pool_address(program_id, vote_account_info.key, pool_info.key)?;
+
+        SinglePool::from_account_info(pool_info, program_id)?;
+
+        check_pool_stake_address(program_id, pool_info.key, pool_stake_info.key)?;
+        let stake_authority_bump_seed = check_pool_stake_authority_address(
+            program_id,
+            pool_info.key,
+            pool_stake_authority_info.key,
+        )?;
+        check_stake_program(stake_program_info.key)?;
+
+        let (_, pool_stake_state) = get_stake_state(pool_stake_info)?;
+        if pool_stake_state.delegation.deactivation_epoch >= clock.epoch {
+            return Err(SinglePoolError::WrongStakeState.into());
+        }
```

Comment:
I might be missing something, but with this check, it makes it impossible for someone to reactivate a stake that was *just* `DeactivateDelinquent`ed in the same epoch. I think it should be safe to remove.

Reply:
i just always err toward being as restrictive as possible to minimize the chances of corner cases. ill relax it to `>`